pkg/network/usm/utils/connection_rollup.go

```diff
@@ -0,0 +1,291 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package utils
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/network/types"
+)
+
+// ConnectionAggregator provides functionality for rolling-up datapoints from
+// different connections that refer to the same (client, server) pair
+type ConnectionAggregator struct {
+	data map[ipTuple]portValues
+}
+
+// NewConnectionAggregator returns a new instance of a `ConnectionAggregator`
+func NewConnectionAggregator() *ConnectionAggregator {
+	return &ConnectionAggregator{
+		data: make(map[ipTuple]portValues, 1000),
+	}
+}
+
+// ipTuple represents a pair of 64-bit IP addresses.
+//
+// note: we chose generic names ("a" and "b") on purpose, since the order of IPs
+// here doesn't align with the notion of "source/destination", "local/remote"
+// or "client/server".
+type ipTuple struct {
```

Comment:
wondering if we want to use ConnTuple here instead
the pros as I see them - 
1. consolidation across types in the project
2. Allowing a future change to introduce pid and netns (for example from TLS hooks)

Reply:
I'm fine with revisiting this in the future, but right now I'd vote against it because:
* We'll go from 256 bits to 384 bits (50% increase) without making use of it any of it;
* The terminology used in the fields is misleading: our `ipTuples` are not structured as source/destination addresses;
* It contains port number fields which won't be used (they have to be stored separately);